boilerplate/app/navigators/navigation-utilities.ts

```diff
@@ -94,7 +95,7 @@ export function useNavigationPersistence(storage: any, persistenceKey: string) {
   // It is selectively enabled in development mode with
   // the following approach. If you'd like to use navigation persistence
   // in production and dev both, remove the __DEV__ and set the state to false
-  const [isRestored, setIsRestored] = useState(!__DEV__)
+  const [isRestored, setIsRestored] = useState(Config.NavigationIsRestored)
```

Comment:
With the above `useNavigationPersistence` config, it would be more like this:

```suggestion
  // somewhere outside of the hook
  function navigationRestoredDefaultState(persistNavigation: typeof Config.persistNavigation) {
    if (persistNavigation === "always") return false
    if (persistNavigation === "dev" && __DEV__) return false
    if (persistNavigation === "prod" && !__DEV__) return false
    // all other cases, disable restoration
    return true
  }

  const [isRestored, setIsRestored] = useState(navigationRestoredDefaultState(Config.persistNavigation))
```